### PR TITLE
fix the bug that parameters are note transposed.

### PR DESCRIPTION
**Pull request by lcy-seso**
fix the bug that when loading the parameter trained by `hsigmoid` and `NCE` layer in inferring, the parameters are not transposed.

### CODE REVIEW

File: hsigmoid/README.md [hunk 1]
```diff
@@ -86,31 +86,30 @@ def ngram_lm(hidden_size, embed_size, dict_size, gram_num=4, is_train=True):
...
+    bias_attr=paddle.attr.Param(name="sigmoid_b"))
+```
+上述代码片段中的 `paddle.layer.mixed` 必须以 PaddlePaddle 中 `paddle.layer.×_projection` 为输入。`paddle.layer.mixed` 将多个 `projection` （输入可以是多个）计算结果求和作为输出。`paddle.layer.trans_full_matrix_projection` 在计算矩阵乘法时会对参数$W$进行转置。
 
 ## 训练阶段
 训练比较简单，直接运行``` python train.py ```。程序第一次运行会检测用户缓存文件夹中是否包含imikolov数据集，如果未包含，则自动下载。运行过程中，每100个iteration会打印模型训练信息，主要包含训练损失和测试损失，每个pass会保存一次模型。
 
 ## 预测阶段
-预测时，直接运行``` python infer.py ```，程序会首先load模型，然后按照batch方式进行预测，并打印预测结果。预测阶段最重要的就是根据概率得到编码路径，然后遍历路径获取最终的预测类别，这部分逻辑如下：
+预测时，在命令行运行 `infer.py --model_path XX`，通过`model_path`指定训练好的模型所在的路径。程序会首先load模型，然后按照batch方式进行预测，并打印预测结果。预测阶段最重要的就是根据概率得到编码路径，然后遍历路径获取最终的预测类别，这部分逻辑如下：
```
Comment (Yancey0623) [inline hsigmoid/README.md]: `python infer.py --model_path ...`

File: nce_cost/README.md [hunk 2]
```diff
@@ -73,22 +73,22 @@ NCE 层的一些重要参数解释如下：
...
-
-	```python
-	prediction = paddle.layer.fc(
-	    size=dict_size,
-	    act=paddle.activation.Softmax(),
-	    bias_attr=paddle.attr.Param(name="nce_b"),
-	    input=hidden_layer,
-	    param_attr=paddle.attr.Param(name="nce_w"))
-	```
+1. 在命令行运行 `infer.py --model_path XX`，通过`model_path`指定训练好的模型所在的路径。
```
Comment (Yancey0623) [inline nce_cost/README.md]: `python infer.py --model_path ...`

File: hsigmoid/network_conf.py [hunk 3]
```diff
@@ -30,18 +31,27 @@ def ngram_lm(hidden_size, embed_size, dict_size, gram_num=4, is_train=True):
             initial_std=1. / math.sqrt(embed_size * 8), learning_rate=1))
 
     if is_train == True:
-        cost = paddle.layer.hsigmoid(
+        return paddle.layer.hsigmoid(
             input=hidden_layer,
             label=target_word,
-            num_classes=dict_size,
```
Comment (Yancey0623) [inline hsigmoid/network_conf.py]: According to the code: https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/gserver/layers/HierarchicalSigmoidLayer.cpp#L27, maybe `num_classes` is necessary?

File: nce_cost/README.md (hunk 2 above)
Comment (lcy-seso) [reply nce_cost/README.md]: The develop branch has been updated. If the parameter `num_classes` is not set, it is automatically determined by the size of the input`label` https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/trainer_config_helpers/layers.py#L2344.

This may cause the problem if the size of `label` is not correctly set (If the size of `label` is not correctly set, PaddlePaddle will not throw an error.), maybe explicitly set the `num_classes` is better.

File: hsigmoid/README.md (hunk 1 above)
Comment (lcy-seso) [reply hsigmoid/README.md]: Done.

File: hsigmoid/network_conf.py (hunk 3 above)
Comment (lcy-seso) [reply hsigmoid/network_conf.py]: Done.

File: nce_cost/README.md (hunk 2 above)
Comment (lcy-seso) [reply nce_cost/README.md]: Done.

**Review by Yancey0623 — APPROVED** [review]
LGTM!
